packages/datadog-instrumentations/src/kafkajs.js

```diff
@@ -103,39 +107,72 @@ addHook({ name: 'kafkajs', file: 'src/index.js', versions: ['>=1.4'] }, (BaseKaf
     const run = consumer.run
 
     const groupId = arguments[0].groupId
-    consumer.run = function ({ eachMessage, ...runArgs }) {
-      if (typeof eachMessage !== 'function') return run({ eachMessage, ...runArgs })
-
+    consumer.run = function ({ eachMessage, eachBatch, ...runArgs }) {
       return run({
-        eachMessage: function (...eachMessageArgs) {
-          const innerAsyncResource = new AsyncResource('bound-anonymous-fn')
-          return innerAsyncResource.runInAsyncScope(() => {
-            const { topic, partition, message } = eachMessageArgs[0]
-            consumerStartCh.publish({ topic, partition, message, groupId })
-            try {
-              const result = eachMessage.apply(this, eachMessageArgs)
-              if (result && typeof result.then === 'function') {
-                result.then(
-                  innerAsyncResource.bind(() => consumerFinishCh.publish(undefined)),
-                  innerAsyncResource.bind(err => {
-                    if (err) {
-                      consumerErrorCh.publish(err)
-                    }
-                    consumerFinishCh.publish(undefined)
-                  })
-                )
-              } else {
+        eachMessage: typeof eachMessage === 'function'
+          ? function (...eachMessageArgs) {
+            const innerAsyncResource = new AsyncResource('bound-anonymous-fn')
+            return innerAsyncResource.runInAsyncScope(() => {
+              const { topic, partition, message } = eachMessageArgs[0]
+              consumerStartCh.publish({ topic, partition, message, groupId })
+              try {
+                const result = eachMessage.apply(this, eachMessageArgs)
+                if (result && typeof result.then === 'function') {
+                  result.then(
+                    innerAsyncResource.bind(() => consumerFinishCh.publish(undefined)),
+                    innerAsyncResource.bind(err => {
+                      if (err) {
+                        consumerErrorCh.publish(err)
+                      }
+                      consumerFinishCh.publish(undefined)
+                    })
+                  )
+                } else {
+                  consumerFinishCh.publish(undefined)
+                }
+
+                return result
+              } catch (e) {
+                consumerErrorCh.publish(e)
                 consumerFinishCh.publish(undefined)
+                throw e
               }
-
-              return result
-            } catch (e) {
-              consumerErrorCh.publish(e)
-              consumerFinishCh.publish(undefined)
-              throw e
+            })
+          }
+          : eachMessage,
+        eachBatch:
+          typeof eachBatch === 'function'
+            ? function (...eachBatchArgs) {
```

Comment:
This function and the one on line 113 are super similar. Can you instead come up with a common function to DRY the code?

Reply:
Updating this to use abstracted code to remove any duplication 👍